src/api/Tron.js

```diff
@@ -491,10 +491,31 @@ export const extractBandwidthInfo = (
 };
 
 export const getTronResources = async (
-  acc: Object,
+  acc?: Object,
   txs: TrongridTxInfo[],
   cacheTransactionInfoById: { [_: string]: TronTransactionInfo }
 ): Promise<TronResources> => {
+  if (!acc) {
+    return {
+      frozen: {
+        bandwidth: undefined,
+        energy: undefined,
+      },
+      delegatedFrozen: {
+        bandwidth: undefined,
+        energy: undefined,
+      },
+      votes: [],
+      tronPower: 0,
+      energy: BigNumber(0),
+      bandwidth: extractBandwidthInfo(),
+      unwithdrawnReward: BigNumber(0),
+      lastWithdrawnRewardDate: undefined,
+      lastVotedDate: undefined,
+      cacheTransactionInfoById: {},
+    };
```

Comment:
I think you don't have to modify the `getTronResources` method, you can just define this data as a constant somewhere, maybe in `utils.js`.

Reply:
As you prefer 👍 

It felt to me that it was a bit more logical to always call the `getTronResources` no matter what use case you have (with account or not) to actually getting `tronResources` (like an unfailable API pretty much). Kinda like the opposite of "separation of concerns", a "reunion of concerns" if that exists. 😆 

I tried to avoid modifying the api too much by letting `txs` & `cacheTransactionInfoById` mandatory to avoid being too confusing. 

Do you think I can mix your approach with this one by declaring those default `tronResources` in `utils.js` and returning it in `getTronResources` ? If not, I'll make the change asap 👍

Reply:
Actually it's the call that bugs me ^^ `await getTronResources(null, [], {})` with these dummy params.
I'd rather keep the current expectations of the API method, but up to you.

Reply:
Yeah I agree... It was that or making `txs` & `cacheTransactionInfoById` optional which felt to me like it was super confusing when using the method since it's more a question of "all or nothing" for the parameters 😕 

I think I found a better solution from this (stackoverflow)[https://stackoverflow.com/a/36274962], just don't know if it's doable with flow yet, I have to check.

Reply:
Ok so I found a solution for Flow but my IDE is getting crazy for whatever reason... I tried in the Flow online validator which declare no issue... I don't know if it's just VSCode getting a false positive or not 😞 

Do you see any problem in [this](https://flow.org/try/#0PTAEFpKrQEQeQHIHIAqoAy94GlQFVFYBRAJVFQAkBJAZVGtWXoCl9b0BZeAYR2sQBxUAEEiFAJoAFYrQBQICNGihB8CuowDioABy7QAQjkmALgE8ADgFNQAIQCWAc0QBXALYAjawCdQAXlAAOw9vHwBuMytbVB8AeyCnHwcAE1QAD2oggDM4gNAAQyDzSLkLGzhrABtrJwLTaxSAMXiAL2sgrNz8gG8AHzlQQvc41yDTAC57ZzcvXwAaOT6AX1Ly2xa49s6cvMD+weHR8anHF1CFw+t0ywcfRpFJuHrrRZW16NAANTiG3oGhgUUil7gBnUFTUGmZKJRZDABuv2sPGOTxCcx8b1WUQqdiKKQA7qlTAALLp7UAHIbZe7WfCgxqnGYXTGHGnWawYBzuBxPM6zMJw0B1BxBRr0xnTc4YoUisUpLk8vnMmVLbFlT6xBKxIqggoAY1MDgS5P+bI5U3RgsOniqcX1AGsBb5LSyhUTSSCCgSAPoFEZjNFuw5jdnWdp+gMnYLB944mLxIKkayg0Y+fUps3UtodKY9Q5DTz4okpUlTAD8m225KFQw6vic5grVY6NcOyyFKWqtRezRzQTzBdARaChOJJIrsG7dQafa2rd2tdA9Z8jcn097LZ2uSFHcOiIaEO+SIA2gBdIXQhJSOIEl0x1V1sWrptS52swvF8enL+lsmLkMgg9EkvQJJNrAJAofBSJlpWtIYqgKKEAHVeRAnxvXAyDoNgF5JxeIVEKhH5Z1whp8IaIV9QNElrB1II9UNY1tziOxzGoGDKVAE8fUhaFRScM8pi1IJ6MYo0TV2UA9zjOR9QSKFlyCeSuwAVl0HhaMdKYAAooRhJwAEoAgAPlAfSBPyUFTPM0p5IY0xhWsJ4dKKcx5kKYoPLc4z-DMtz8lc+ZPHmfVfLM5B5NcKLkDshTHKcZyRJEfUovGRBnIJOIfAdXSfJsqR4h5BkAB43LMwIdPhcLQEKuJiusAA6ME4iqeFrCqwzwlAEx7MUxLTHwIC0NArCoM41zgR8PiDJquqGpK-kWQq0BJpBOaiocBlmpTVr2p0paMSaIIdIABkMwzevit8WWOwKgk8V0MRqh64oc5d0mhA1TDxUcS1JU1KoKGr8yGIZ7lMVwfCCSkhzDCVOMOsJjrOwylzDRVeVg98UfOpc5XFBlEZVZGTrxocCYVbksZuo6ybR9tImWEwFDAZQVAQFB0CwXBRDseAvh0Kg6AYJhWHYLheH4IRRHEVBpFkVmlHZ8A4CQNBMGwPARH5wWKBoehGGYUA2A4UBuD4ARhDEWBJBkeRFBVyA1a5zXeZ1gWhYN0XjdNyXLZlm27cVlnQ5MLt9UQ+5QGyMYmISJzTBE5NUyhjMdMMqZ5q26wSuT3a05TEzIkUFMqlFUxwBSLaCltaxwDFT7wHLsVgjicB7gjqPrDkLuoNsWPlIkmGBvz1P0w6w4DX1KZ4E8AAraxDUvdIjxEpJUgyclzyomi6IwhjvuY8k2I4vNuN48z+MSISKETMSj8k7pmcz2rNtKse0wzUFi6VsuK6rjXOuDdriVxbrYII7dO5L27nIa4lhsqOSQuYZSMc47D0Tp-QuOkp6pXLLPBeS9TAr1BPgu+CQN5pEyLsHehxqL6log-eOLFT4pDIT0C+M0BK3xEkw4epoX5Z3frnLBE8f6wyGA4bIq1DDT2MqDMGEMoYwwUWDGO-ZBxqLUSOMcf4phjC7NkUUjQlxqJXGuUABjrBGPlKY6Sdiuw1BnI0LcmitGfj+t+Sxo5rHGJSHYp8DZXxWJsSYocQw9zuIPCmKYtD3FXiCDeO801QCnTseY18SNfC4wZu4nR-0JwfS+oaX6uiAa7AznYsYwFRrJmwsTOC2T6Z2KIqYVCnoMJgTqeNMi1h9E+NCf48JoBWkkUaL0-phi-F2PoYwg+4lj67FYXmSJYNsQRJMEMPqjkaTziCKUgp+QBquVSh5AARLs7YJ5TpnjOR5EJfiup0OupcjoxBnyNiOc5HBajp5LjOdPVEPoWqF0aq8oIPoiyIWUtYH0uQfA+gyXcocDz5SHCec896jieyzi3Ac8cXzTA-LBn8ocZzsXOJSHC-skKCjQozHC7KtLPF-mRWo1FjR0WRC2ddClm5+zvKCYS4lgJTlksBYGYFBcJ6NT5bOalezaX0thfCxFHzzBsrBhylIXLNmgG2UpFSDwpopmsoEDoRr1KaSXg6E5+pGpAhBKajFPL3oJIyqYLKOVAaFEgryTBiYUppVMB6r1tqLVxC7CkEQJrwRPNdYpapI1OljWgvkb0BR-UDSGjUlN3ToI6QjVGmNTq42lATY5DJ+R3WZWyg6ckjUMmY1MI1HkIRQQ6RrZ6utDaMkI3jfq66+SCXms+hhEpv5ym5E7YmUNPbdgusHe9cFvQhzDr0eovZ+K-zDPLBI9xJKozKkaT4FG4Lt2kjBTSqFRQMy5IPXWG4dwHhoggs8BoOlz2TpJI2p99wfRGncNYe97jmbuMmb42xQ4MlTHBYKl8u790Pv9KibGt0TpwfVVexVN6YUgYPfA590bX0EnfR1TDQTf23H-YB4DATpLDIg4M3c5al2KTlS4-sq7tHfqmBxuc2wL0kkQxwlDgY0N0x0vxvF37jJga0Ux6Z0H1V8Y3LigV6qRNHHE7TUmUm1Ocb2fBxscnGPeKmVBiJrGDUJKSb4fIWTT0nWOeC8567L1ifGHc1JF1rPXWiWaxOdrzkBe8+eQyraCiWB0p1GyOlVGFFjUeeEjVol+iS0uaJKIdMpbS8G3cvm9UGtae09CmF80pF6em1KjVEKHlMD6XN3oAPch7movdYpSO9LtbVl4UJGvJua7Ry6ajFPylY4oVA8AEBTHuFUcwoAE6klsBK8Yi3CGGlAASWiMMEmUK2w4KoVRQCWHiPCVIthSRbVAKKXIjVMX9WchgJCpgxmVZeIFUwq9hKJkoVvGhZ5X7liq35JDbHHKtPgJYatq8wWihSDFjh6xpI1WR-4dHoAzljPaWKcEKJxjjtMGcgd4NnLKJGS9qHoA92Q8sLKj7rojvcuksz4rL23sg9AF96ye6BrPeIkid7H7ufGTG40VjSjoZg4yUudzJJ0b9iXNJxXQ4AtLls7eS47LhodPKxBcaS4SuDa6frnCBEhyjMF70pcsz966kfiw9inFbd8MWbkVh1O909FWdiZmchR6JhTl-DqTyA8JCD9gkIR3Q9JUD9K9O3uY9Jzj+PBPHZuKRVGDFQHkQw-gVTx1b3HkTyZ+iqMZAF5KTLC6kAA) ?